src/main/java/io/neonbee/cluster/ClusterManagerFactory.java

```diff
@@ -31,8 +36,35 @@ protected String getDefaultConfig() {
         @Override
         public Future<ClusterManager> create(NeonBeeOptions neonBeeOptions) {
             String effectiveConfig = getEffectiveConfig(neonBeeOptions);
-            return succeededFuture(new HazelcastClusterManager(
-                    new ClasspathXmlConfig(getSystemClassLoader(), effectiveConfig, System.getProperties())));
+
+            return Future.future(promise -> {
+                Config config;
+                if (exitsInClasspath(effectiveConfig)) {
+                    config = new ClasspathXmlConfig(getSystemClassLoader(), effectiveConfig, System.getProperties());
+                } else {
+                    try {
+                        config = new FileSystemXmlConfig(effectiveConfig, System.getProperties());
+                    } catch (FileNotFoundException e) {
+                        throw new RuntimeException(e);
```

Comment:
Would it make sense not to throw an exception here, but to return a failed future describing that the configuration was not found in both the classpath and the filesystem?

Reply:
I wrapt the Exception in a RuntimeException to fail the promise. I could add an additional descriptive message to the RuntimeException.
I also could fail the promise at this point to avoid warping the FileNotFoundException in a RuntimeException. 
```
try {
    config = new FileSystemXmlConfig(effectiveConfig, System.getProperties());
} catch (FileNotFoundException e) {
    promise.fail(e);
    return;
}
```
At the moment, I don't see any real advantage of one method over the other.

Reply:
You are right the promise is catching the exception